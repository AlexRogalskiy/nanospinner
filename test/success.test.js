let { createSpinner } = require('../index.js')

let stdout = { out: '' }
stdout.write = (symbols) => {
  stdout.out += symbols
}

it('marks spinner as success', () => {
  stdout.out = ''
  let spinner = createSpinner('#success', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.success()

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})

it('marks spinner as success with message', () => {
  stdout.out = ''
  let spinner = createSpinner('#success', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.success({ text: 'Successful' })

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})

it('marks spinner as success with mark', () => {
  stdout.out = ''
  let spinner = createSpinner('#success', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.success({ mark: 'V' })

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})
