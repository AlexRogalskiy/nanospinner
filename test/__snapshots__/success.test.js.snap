// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marks spinner as success with mark: CI 1`] = `
"[33m-[39m #success
[33m-[39m #success
[33m-[39m #success
V #success
"
`;

exports[`marks spinner as success with mark: Local 1`] = `
"[?25l[1G[33mâ ‹[39m #success[?25l[1G[2K[1G[33mâ ™[39m #success[?25l[1G[2K[1G[33mâ ¹[39m #success[1G[2K[1GV #success
[?25h"
`;

exports[`marks spinner as success with message: CI 1`] = `
"[33m-[39m #success
[33m-[39m #success
[33m-[39m #success
[32mâœ”[39m Successful
"
`;

exports[`marks spinner as success with message: Local 1`] = `
"[?25l[1G[33mâ ‹[39m #success[?25l[1G[2K[1G[33mâ ™[39m #success[?25l[1G[2K[1G[33mâ ¹[39m #success[1G[2K[1G[32mâœ”[39m Successful
[?25h"
`;

exports[`marks spinner as success: CI 1`] = `
"[33m-[39m #success
[33m-[39m #success
[33m-[39m #success
[32mâœ”[39m #success
"
`;

exports[`marks spinner as success: Local 1`] = `
"[?25l[1G[33mâ ‹[39m #success[?25l[1G[2K[1G[33mâ ™[39m #success[?25l[1G[2K[1G[33mâ ¹[39m #success[1G[2K[1G[32mâœ”[39m #success
[?25h"
`;
